src/features/contact/contactverzoek/overzicht/service.ts

```diff
@@ -15,69 +17,104 @@ import {
+    url.searchParams.set("ordering", "-record__data__registratiedatum");
+    url.searchParams.set("pageSize", "10");
+    url.searchParams.set(
+      "data_attrs",
+      `betrokkene__digitaleAdressen__icontains__${query}`
+    );
+  }
 
   return url.toString();
 }
 
 function searchRecursive(urlStr: string, page = 1): Promise<any[]> {
-  //recursive alle pagina's ophalen.
-  //set de page param van de huidige op te halen pagina
   const url = new URL(urlStr);
   url.searchParams.set("page", page.toString());
 
-  return (
-    fetchLoggedIn(url)
-      .then(throwIfNotOk)
-      .then(parseJson)
-      //todo: parsePagination toevoegen??
-      .then(async (j) => {
-        if (!Array.isArray(j?.results)) {
-          throw new Error("expected array: " + JSON.stringify(j));
-        }
-
-        const result: any[] = [];
-        j.results.forEach((k: any) => {
-          result.push(k);
-        });
-
-        if (!j.next) return result;
-        return await searchRecursive(urlStr, page + 1).then((next) => [
-          ...result,
-          ...next,
-        ]);
-      })
-  );
+  return fetchLoggedIn(url)
+    .then(throwIfNotOk)
+    .then(parseJson)
+    .then(async (j) => {
+      if (!Array.isArray(j?.results)) {
+        throw new Error("Expected array: " + JSON.stringify(j));
+      }
+
+      const result: any[] = [...j.results];
+
+      if (!j.next) return result; 
+      const nextResults = await searchRecursive(urlStr, page + 1);
+      return [...result, ...nextResults];
+    });
 }
 
-function search(url: string) {
-  return searchRecursive(url); //todo: sortering???  .then((r) => r.sort(sortKlant));
+export async function search(
+  query: string,
+  gebruikKlantInteractiesApi: Ref<boolean | null>
+): Promise<PaginatedResult<Contactverzoek>[]> { 
+  const url = getSearchUrl(query, gebruikKlantInteractiesApi);
+
+  if (gebruikKlantInteractiesApi.value) {
+    const searchResults = await searchRecursive(url);
+    const enrichedResults = await Promise.all(
+      searchResults.map(async (result: any) => {
+        const klantUrl = ref(result.url);
+        return await getContactverzoekenByKlantUrl(klantUrl);
+      })
+    );
+    return enrichedResults;
+  } else {
+    const searchResults = await searchRecursive(url);
+    return [mapObjectToContactverzoekViewModel({
+      next: null,
+      previous: null,
+      count: searchResults.length,
+      results: searchResults,
+    })]; 
+  }
 }
 
-export function useSearch(params: Ref<SearchParameters>) {
-  const getUrl = () => getSearchUrl(params.value);
-  return ServiceResult.fromFetcher(getUrl, search);
+export async function getContactverzoekenByKlantUrl(klantUrl: Ref<string>) {
+  function getUrl() {
+    const searchParams = new URLSearchParams();
+    searchParams.set("verstrektedigitaalAdres__url", klantUrl.value);
+    return `${klantinteractiesBetrokkenen}?${searchParams.toString()}`;
+  }
+
+  return fetchBetrokkene(getUrl())
```

Comment:
ditigtale adressen heeft een expand op "verstrektDoorBetrokkene.hadKlantcontact.leiddeTotInterneTaken". volgens mij hoef je dan veel minder calls te doen

Reply:
De expand biedt voornamelijk verwijzingen (UUIDs en URLs) naar gerelateerde data in plaats van de volledige gegevens zelf. De wasPartij property die cruciaal is voor de filtering wordt ook geleverd door de enrichBetrokkeneWithKlantContact dus ik heb heel de expand eruit gehaald.

De expand voegt weinig waarde toe omdat het geen volledige data retourneert die voor verrijking nodig is.